arrow-flight/src/encode.rs

```diff
@@ -189,26 +215,30 @@ impl FlightDataEncoder {
         }
     }
 
+    /// Encodes schema as a [`FlightData`] in self.queue.
+    /// Updates `self.schema` and returns the new schema
+    fn encode_schema(&mut self, schema: SchemaRef) -> SchemaRef {
```

Comment:
```suggestion
    /// Updates `self.schema` and returns the new schema
    fn encode_schema(&mut self, schema: &Schema) {
```
Might be a more obvious signature imo

Reply:
changed to     `fn encode_schema(&mut self, schema: &SchemaRef) {`